app/uk/gov/hmrc/trackingconsentfrontend/filters/CookieConsentFilter.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.trackingconsentfrontend.filters
+
+import play.api.mvc._
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+
+class CookieConsentFilter @Inject() (implicit ec: ExecutionContext) extends EssentialFilter {
+  // This filter is not intended to stay forever, but rather to alleviate an issue with a bug, as per PLATUI-3500
+  override def apply(next: EssentialAction): EssentialAction = EssentialAction { requestHeader =>
+    next(requestHeader).map { result =>
+      val modifiedCookies = requestHeader.cookies.map { c =>
+        if ((c.name == "userConsent" || c.name == "mdtpurr") && c.httpOnly) {
+          c.copy(httpOnly = false, maxAge = c.maxAge.orElse(Some(31556926)))
+        } else c
```

Comment:
```suggestion
        if ((c.name == "userConsent" || c.name == "mdtpurr")) {
          c.copy(httpOnly = false)
        } else c
```
I don't think we need these bets because those will just be the default of the Cookie case class, browser doesn't send this metadata to the server

Reply:
I am not sure about `maxAge`, this came from the spike results, so just followed that, but about `httpOnly` I believe that we only want to fix the broken cookies not the correct ones

Reply:
@PatrykRudnicki @oscarduignan worth breaking out `31556926` into a val with a friendly name? Something like `val secondsInAYear = 31556926`?